packages/cli/src/generator.ts

```diff
@@ -144,15 +229,23 @@ export function getNamedImportNodes(
     return node;
   }
 
-  // If the named bindings are a named import, get the import names.
-  const importNames = namedBindings.elements
-    .filter((element) => !element.isTypeOnly)
-    .map((element) => ({
-      name: element.name.text,
-      propertyName: element.propertyName?.text,
-    }));
+  const importNames = getImports(
+    node.moduleSpecifier.text,
+    system,
+    sourceFile.fileName,
+    namedBindings.elements,
+  );
 
-  if (importNames.length === 0) {
+  // If there are no named imports, return the node as is.
+  if (
+    importNames.detected.length === 0 &&
+    importNames.undetected.length === 0
+  ) {
+    return node;
+  }
+
+  // If there are no undetected imports, return the node as is.
+  if (importNames.undetected.length === 0) {
```

Comment:
Hmm, why do we not need to codegen if this is the case?

Reply:
Imports that are detected by `cjs-module-lexer` can be imported as-is by Node.js, so if all imports in the import statement here are detected, this will work without transforming it.

```ts
import { importA, importB, importC } from 'cjs-module';
```

Reply:
Does that apply to Webpack etc too? I ask because before we would transform everything

Reply:
Yes. I tested Webpack and Esbuild (used by Vite), and I assume other tools like Rollup support the same thing too.